### PR TITLE
Enhance `update_platform` to support more use cases, add more consistency

### PR DESCRIPTION
**Pull request by emiliom**
Starts to address #740. Initial updates only:
- Enables the use case of adding a simple lat-lon point as the external platform data. Previously `update_platform` failed with that use case.
- Attributes for the target variables are now set from `sonarnetcdf_1.yaml_dict["variable_and_varattributes"]["platform_var_default"]`
- Made some simplifications and generalizations in `update_platform`
- Previously `update_platform` set all missing target variables to nan except `water_level`, which was set to 0. I didn't find any reason for it, so I changed it to set it to nan like all other variables (that made the code cleaner). That broke `tests/echodata/test_echodata.py::test_update_platform`, but for bogus reasons. I fixed the test.

### CODE REVIEW

File: echopype/echodata/echodata.py [hunk 1]
```diff
@@ -620,73 +621,32 @@ def update_platform(
...
-                    mapping_search_variable(
-                        extra_platform_data,
-                        ["water_level", "WATER_LEVEL"],
-                        default=platform.get("water_level", np.zeros(num_obs)),
-                    ),
-                ),
-            }
-        )
+        num_obs = len(extra_platform_data[time_dim])
+        for platform_var, sourcenames in platform_vars_sourcenames.items():
```
Comment (leewujung) [inline echopype/echodata/echodata.py:631]: This is so much cleaner! 🥳

**Comment by leewujung** [comment]
I found it very bizarre that my changes from the already merged PR #714 show up in this PR, even though the target is `dev`. Also had a couple of comments on the initial changes.

**Comment by emiliom** [comment]
> I found it very bizarre that my changes from the already merged PR https://github.com/OSOceanAcoustics/echopype/pull/714 show up in this PR, even though the target is dev. Also had a couple of comments on the initial changes.

Sigh. I have no idea why that's showing up. My PR originally contained just two changed files, `echodata.py` and `test_echodata.py`. Sometimes git and github are mysterious.

File: echopype/tests/echodata/test_echodata.py [hunk 2]
```diff
@@ -472,7 +472,7 @@ def test_update_platform(update_platform_samples):
 
     ed = echopype.open_raw(raw_file, "EK80")
 
-    updated = ["pitch", "roll", "latitude", "longitude", "water_level"]
+    updated = ["pitch", "roll", "latitude", "longitude"]
     for variable in updated:
         assert np.isnan(ed.platform[variable].values).all()
```
[the remark this answers is not in the document]
Comment (leewujung) [reply echopype/tests/echodata/test_echodata.py:477]: Ah, I see. Thanks for the explanation!

**Review by leewujung — APPROVED** [review]

**Comment by emiliom** [comment]
Thanks for your review. I suggest we keep this PR unmerged for now, until we decide if anything else from #740 should make it into 0.6.1; for example, adding variable `history` attributes should be easy.

One other thing we (I?) could *probably* add in time for 0.6.1 is a test for the fixed-location use case.

**Comment by emiliom** [comment]
This PR is now "clean" of the git/github mess! It no longer includes the extraneous files from a different PR, and it's back at just the two files being targeted.

**Comment by emiliom** [comment]
Added commit that adds `history` attribute to variables that are inserted

**Comment by emiliom** [comment]
Added commit that adds a test for the fixed-location use case. To do that, I overhauled the `test_update_platform` test, including the pytest fixture/parameters associated with it.

**Comment by emiliom** [comment]
@leewujung could you take one last look at this PR, now that I've added:
- `history` attribute to variables that are inserted
-  A test for the fixed-location use case. To do that, I overhauled the `test_update_platform` test, including the pytest fixture/parameters associated with it.

I'm done with the enhancements I proposed for 0.6.1.

File: echopype/tests/echodata/test_echodata.py [hunk 3]
```diff
@@ -466,17 +440,68 @@ def test_nan_range_entries(range_check_files):
...
+                        "saildrone",
+                        "SD2019_WCS_v05-Phase0-D20190617-T125959-0.raw",
+                ),
+                (
+                        "saildrone",
+                        "saildrone-gen_5-fisheries-acoustics-code-sprint-sd1039-20190617T130000-20190618T125959-1_hz-v1.1595357449818.nc",  #noqa
+                ),
+        ),
+        (
+                "fixedlocation",
```
Comment (leewujung) [inline echopype/tests/echodata/test_echodata.py:461]: I just have a super minor suggestion here: `fixedlocation` --> `fixed-location` to be consistent with `external-trajectory`

Comment (leewujung) [inline echopype/tests/echodata/test_echodata.py:461]:
```suggestion
                "fixed-location",
```
to be consistent with `external-trajectory`

File: echopype/tests/echodata/test_echodata.py [hunk 4]
```diff
@@ -466,17 +440,68 @@ def test_nan_range_entries(range_check_files):
...
-
-    ed = echopype.open_raw(raw_file, "EK80")
+@pytest.mark.parametrize(
+    ["ext_type", "sonar_model", "updated", "path_model", "raw_path", "platform_data"],
+    [
+        (
+                "external-trajectory",
+                "EK80",
+                ("pitch", "roll", "longitude", "latitude"),
+                "EK80",
```
Comment (leewujung) [inline echopype/tests/echodata/test_echodata.py:450]: It seems that the entire block is shifted 1 tab more than needed to the right. Not crucial at all, just the line overflows my screen.

File: echopype/tests/echodata/test_echodata.py [hunk 5]
```diff
@@ -496,7 +521,7 @@ def test_update_platform(update_platform_samples):
         np.count_nonzero(
             ed.platform["time1"] < ed.beam["ping_time"].min()
         )
-        == 1
+        <= 1
```
Comment (leewujung) [inline echopype/tests/echodata/test_echodata.py:524]: This is tied to this particular data file and not generally true, right?

**Review by leewujung — APPROVED** [review]
@emiliom : the changes look good. I like the new tests that are very specific on the use cases. I just have one question in the last comment. Otherwise this seems ready to go!

File: echopype/tests/echodata/test_echodata.py [hunk 6]
```diff
@@ -1,3 +1,4 @@
+from pathlib import Path
```
Comment (leewujung) [inline echopype/tests/echodata/test_echodata.py:1]: I think this is redundant.

File: echopype/tests/echodata/test_echodata.py [hunk 7]
```diff
@@ -466,17 +440,68 @@ def test_nan_range_entries(range_check_files):
...
     for variable in updated:
         assert np.isnan(ed.platform[variable].values).all()
 
-    extra_platform_data = xr.open_dataset(extra_platform_data_file)
+    if ext_type == "external-trajectory":
+        extra_platform_data_file_name = platform_data[1]
+        extra_platform_data = xr.open_dataset(
+            test_path[path_model] / platform_data[0] / extra_platform_data_file_name
+        )
+    elif ext_type == "fixedlocation":
```
Comment (emiliom) [inline echopype/tests/echodata/test_echodata.py:493]:
```suggestion
    elif ext_type == "fixed-location":
```

File: echopype/tests/echodata/test_echodata.py (hunk 4 above)
Comment (emiliom) [reply echopype/tests/echodata/test_echodata.py:450]: Yeah, I saw that. I think PyCharm did it. I figured pre-commit would've fixed it, but it looks like it didn't.

File: echopype/tests/echodata/test_echodata.py (hunk 6 above)
Comment (emiliom) [reply echopype/tests/echodata/test_echodata.py:1]: Right! It's not used. I'll remove it.

Comment (emiliom) [inline echopype/tests/echodata/test_echodata.py:1]:
```suggestion
```

**Pull request merged**